common/util.go

```diff
@@ -407,6 +408,24 @@ func ValidateCronSchedule(cronSchedule string) error {
 	return nil
 }
 
+// GetBackoffForNextCronSchedule calculates the backoff time for the next run given
+// a cronSchedule and current time
+func GetBackoffForNextCronSchedule(cronSchedule string, nowTime time.Time) time.Duration {
```

Comment:
plz do not put logic in the util file / package.

util does not provide good indication about anything
try time or cron package / file name